integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java

```diff
@@ -18,24 +18,34 @@
 import java.util.ArrayList;
 import java.util.Collection;
 
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+
 import org.ocpsoft.common.spi.ServiceEnricher;
 import org.ocpsoft.logging.Logger;
+import org.ocpsoft.rewrite.servlet.spi.ContextListener;
 import org.springframework.web.context.support.SpringBeanAutowiringSupport;
 
 /**
  * {@link ServiceEnricher} implementation for Spring.
  * 
  * @author Christian Kaltepoth
  */
-public class SpringServiceEnricher implements ServiceEnricher
+public class SpringServiceEnricher implements ServiceEnricher, ContextListener
 {
 
    private final Logger log = Logger.getLogger(SpringServiceEnricher.class);
+   private static ServletContext context;
```

Comment:
What happens, if this class runs in a traditionally deployed war file in a servlet container?

If there were multiple applications deployed, the ServletContext instances could overwrite each other and possibly leak into the wrong application?

Reply:
You mean in Servlet Container where someone has put shared dependencies in the servlet container classpath? Or in an EAR type container where JARs and dependencies are shared between WARs? I guess that could be a concern. Let me think about that.